src/compute/src/render.rs

```diff
@@ -506,6 +506,13 @@ where
                 if let Some(logger) = compute_state.compute_logger.clone() {
                     errs.stream.log_dataflow_errors(logger, idx_id);
                 }
+                // If the dataflow is set to expire at some point, we'll need to make sure that the frontier
+                // does not advance past the expiration time. Otherwise, we might write down incorrect
+                // data.
+                if let Some(timestamp) = self.expire_at.as_option().copied() {
+                    oks.expire_at(timestamp);
```

Comment:
We could assign to `oks` to make sure that we're observing the frontier before any downstream operators.